### ISSUE TITLE
Transaction DELETE args do not match url params

### ISSUE THREAD

**Issue by Mahir-Sparkess** [stac-utils/stac-fastapi#272]
The arguments in `delete_collection` and `delete_item` do not match the URL endpoint. This leads to the parameters being passed to the `kwargs` instead of the args: `item_id` and `collection_id`, as well as gives a `TypeError: missing required positional argument`

Solution: 
change the endpoint url parameters: `itemId -> item_id` and `collectionId -> collection_id`,
OR
change the method argument names: `item_id -> itemId` and `collection_id -> collectionId`

https://github.com/stac-utils/stac-fastapi/blob/27e134589107654920c2f1dba54773c8c85d4e1a/stac_fastapi/types/stac_fastapi/types/core.py#L59-L64

**Comment by geospatial-jeff** [comment]
I don't think this is the case with the current backend implementations contained in the repo
- `BaseTransactionsClient.delete_item` takes arguments of `item_id` and `collection_id` (https://github.com/stac-utils/stac-fastapi/blob/master/stac_fastapi/types/stac_fastapi/types/core.py#L60)
- The delete item endpoint (`/collections/{collectionId}/items/{itemId}`) takes an `ItemUri` request model (https://github.com/stac-utils/stac-fastapi/blob/master/stac_fastapi/extensions/stac_fastapi/extensions/core/transaction.py#L106)
- `ItemUri` dumps those variables into `item_id` and `collection_id` (https://github.com/stac-utils/stac-fastapi/blob/master/stac_fastapi/api/stac_fastapi/api/models.py#L73).
- `item_id` and `collection_id` are injected into the `delete_item` call as key word arguments (https://github.com/stac-utils/stac-fastapi/blob/master/stac_fastapi/api/stac_fastapi/api/routes.py#L33).

Also all of the item deletion test cases are passing on master.

I could see this being a problem if the specific backend implementation of `BaseTransactionsClient` requires argument specific parameters that are not being threaded through by the request model (in this case `ItemUri`).  Would be interested in knowing more about your particular use case to see why its breaking.

This all being said, I do think its probably simpler to align the path parameters used by fastapi with the arguments accepted by the backend clients, and that would be a change I'd want to do across all endpoints, not just for item deletion.

**Comment by Mahir-Sparkess** [comment]
I see, I wonder why when implementing transactions I would get `TypeError: missing positional arguments`. As stated in my issue, my solution to that was to rename the variable to match the URI, but I also noticed that the parameter in the `/docs` got passed to the kwargs instead of the related arg.

This issue might be irrelevant now with issue #274.

Thank you for your response.
